### PR TITLE
fix: skip automatic form generation when datastore is not enabled

### PR DESCRIPTION
**Pull request by awinberg-aws**
#### Description of changes

Updates the uibuilder plugin to use a helper recently exposed by the API category to confirm if an application has datastore enabled. Forms that are auto-generated by uibuilder are not compatible with customer applications without datastore being enabled, so this is now being incorporated into the logic for determining if they should be generated or skipped.

#### Description of how you validated changes
ran `amplify-dev pull` against a test project with datastore disabled to confirm that no auto-generated forms were created, and again with datastore enabled to confirm that the auto-generated forms are still created when opted-in.
#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by letsbelopez — DISMISSED** [review]
LGTM

**Review by lyonsbp — APPROVED** [review]

**Review by lyonsbp — APPROVED** [review]

**Review by letsbelopez — APPROVED** [review]
re-approving after merge conflicts resolved

**Review by cwoolum — APPROVED** [review]

**Review by kuhlmanp — APPROVED** [review]

**Pull request merged**
